### PR TITLE
Add tags to examples frontmatter

### PR DESCRIPTION
**Pull request by Taha-Hassan-Git**
Better filtering in the examples app, see gif below.

Drive-by fixes: 
- removed a duplicate shape meta example
- Speech bubble text was overflowing and needed the width to be set
- Sticker (bindings) no longer snaps
- typos

2024-06-12 at 17 06 17 - Jade Woodpecker

### Change Type

- [ ] `sdk` — Changes the tldraw SDK
- [ ] `dotcom` — Changes the tldraw.com web app
- [x] `docs` — Changes to the documentation, examples, or templates.
- [ ] `vs code` — Changes to the vscode plugin
- [ ] `internal` — Does not affect user-facing stuff

- [ ] `bugfix` — Bug fix
- [ ] `feature` — New feature
- [x] `improvement` — Improving existing features
- [ ] `chore` — Updating dependencies, other boring stuff
- [ ] `galaxy brain` — Architectural changes
- [ ] `tests` — Changes to any test code
- [ ] `tools` — Changes to infrastructure, CI, internal scripts, debugging tools, etc.
- [ ] `dunno` — I don't know

### Test Plan

1. Type a lot of text in a speech bubble shape and move the tail
2. Text should stay static

### Release Notes

- Improve filtering of examples

### CODE REVIEW

File: apps/examples/src/examples/ui-components-hidden/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/ui-components-hidden/README.md:5]: "hide ui" maybe as separate words

File: apps/examples/src/examples/yjs/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/yjs/README.md:6]: i might suggest making this two words "multi" and "player" just in case someone types it in like `multi-player`

File: apps/examples/src/examples/readonly/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/readonly/README.md:6]: read only as separate words just in case someone types it as "read-only"

File: apps/examples/src/examples/only-editor/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/only-editor/README.md:6]: "bare" "bones" as separate words

File: apps/examples/src/examples/meta-on-create/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/meta-on-create/README.md:6]: i wouldn't include `getInitialMetaForShape` - feels out of place...

File: apps/examples/src/examples/meta-on-change/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/meta-on-change/README.md:6]: i wouldn't include specific API terms like this - it doesn't fit

File: apps/examples/src/examples/local-storage/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/local-storage/README.md:6]: rm `createTlStore`

File: apps/examples/src/examples/interactive-shape/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/interactive-shape/README.md:6]: why "todo"?

File: apps/examples/src/examples/inline/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/inline/README.md:6]: i would put "auto" as a separate term

File: apps/examples/src/examples/infer-dark-mode/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/infer-dark-mode/README.md:6]: heh, let's remove "tldraw" :)

File: apps/examples/src/examples/hosted-images/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/hosted-images/README.md:7]: let's remove `registerExternalAssetHandler`

File: apps/examples/src/examples/hide-ui/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/hide-ui/README.md:5]: two words: "hide" and "ui"

File: apps/examples/src/examples/force-mobile/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/force-mobile/README.md:6]: maybe just "force" and "mobile"

File: apps/examples/src/examples/external-ui/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/external-ui/README.md:5]: let's rm `useState`

File: apps/examples/src/examples/exploded/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/exploded/README.md:6]: "exploded" probably isn't needed
likewise, let's remove `TldrawEditor` - it wouldn't fit with keywords

File: apps/examples/src/examples/error-boundary/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/error-boundary/README.md:6]: ditto on specific API terms

File: apps/examples/src/examples/editor-focus/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/editor-focus/README.md:6]: let's remove the API terms

File: apps/examples/src/examples/editable-shape/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/editable-shape/README.md:6]: rm the API terms

File: apps/examples/src/examples/drag-and-drop/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/drag-and-drop/README.md:6]: rm API terms

File: apps/examples/src/examples/custom-ui/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/custom-ui/README.md:6]: "hide" and "ui"

File: apps/examples/src/examples/custom-tool/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/custom-tool/README.md:6]: "state" and "node"

File: apps/examples/src/examples/custom-renderer/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/custom-renderer/README.md:5]: "ctx" → "context"
"htmlcanvas" → "html" and "canvas"

File: apps/examples/src/examples/custom-actions-menu/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/custom-actions-menu/README.md:5]: "drop" and "down"

File: apps/examples/src/examples/context-toolbar/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/context-toolbar/README.md:6]: let's rm the API term

File: apps/examples/src/examples/changing-default-colors/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/changing-default-colors/README.md:6]: heh, we prbly want "color", at least in addition :)

File: apps/examples/src/examples/canvas-events/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/canvas-events/README.md:6]: rm `editor.on()`

File: apps/examples/src/examples/bounds-snapping-shape/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/bounds-snapping-shape/README.md:6]: playing card, while accurate, feels strange

File: apps/examples/src/examples/basic/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/basic/README.md:6]: i'd add maybe "intro", "simple", "quick", "start",

File: apps/examples/src/examples/api/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/api/README.md:6]: - rm useEditor
- `onMount` → `mount`

File: apps/examples/src/ExamplePage.tsx
Comment (mimecuvalo) [inline apps/examples/src/ExamplePage.tsx:88]: this wasn't a problem you introduced but one thing we haven't gotten around to fixing here is splitting up the `filterValue` into separate terms and searching on each term.

so, if i filter for a "add tool" it won't find the "Add a tool to the Toolbar" example because i didn't include the "a".

if you want, it'd be great to do a search on each term separately!

File: apps/examples/src/examples/custom-components/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/custom-components/README.md:6]: i've been saying to remove the custom API terms but _maybe_ this is the one exception :)

**Review by mimecuvalo — COMMENTED** [review]
great cleanup and addition! thanks!

high-level comment is to improve the filtering mechanism to work on each term (see my comment below).

a _followup_ PR by the way, if you're interested (since I saw you found a bunch of typos, thx!) is taking a look at this library and intergrating it: https://alexjs.com/
i haven't gotten the chance to set it up for our repo yet but it would be great to add that as well!

File: apps/examples/src/examples/changing-default-colors/README.md
Comment (Taha-Hassan-Git) [reply apps/examples/src/examples/changing-default-colors/README.md:6]: I think that's covered by the title, it's in the keywords as a catch-all

**Comment by steveruizok** [comment]
Is this good to go?

**Comment by Taha-Hassan-Git** [comment]
Yeah, was just waiting for a sign-off from mime

File: apps/examples/src/examples/infer-dark-mode/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/infer-dark-mode/README.md:6]: mm, instead of "props" - let's do theme/light/dark

**Review by mimecuvalo — APPROVED** [review]
looks good - thank you! just one last tweak but good to go

**Pull request merged**
